app/__init__.py

```diff
@@ -44,10 +44,8 @@ def add_cors_header(response):
 # Reject request from non allowed origins
 @app.before_request
 def validate_origin():
-    if 'Origin' not in request.headers:
-        logger.error('Origin header is not set')
-        abort(make_error_msg(403, 'Not allowed'))
-    if not re.match(ALLOWED_DOMAINS_PATTERN, request.headers['Origin']):
+    if 'Origin' in request.headers and \
+        not re.match(ALLOWED_DOMAINS_PATTERN, request.headers['Origin']):
```

Comment:
This is not the same behaviour as the previous implementation. This just fails if an origin header is present but not in the allowed list. But we want that the request fails, if no origin header is present, this would be possible with your change.